### PR TITLE
added support for train_test_split of s3 bucket as well as major API …

### PR DESCRIPTION
**Pull request by conorw8**
## Primary Feature Update
Added support for splitting dataset in s3 into train, val, and test sets. The user has the option of splitting labeled or unlabeled datasets by passing in a s3 url hosting the data and an optional s3 url hosting the corresponding labels. Additionally, the user has the option of just splitting the dataset into train and val sets by just passing in a train_split argument. However, the user can also pass in a val_split parameter which will subsequently split the val set into val and test sets.

Example usage:
```
data_url = "s3://ml-data-split-test/voc_2012_subset/images/"
labels_url = "s3://ml-data-split-test/voc_2012_subset/annotations/"
ml_dronebase_data_utils.s3.split_dataset(data_url, train_split=0.6, val_split=0.5, labels_url)
```

## Secondary Update
Made some relatively major cleaning of the s3 API by formatting the naming schemes (e.g., there was an upload_{file, dir} but a download_s3_{file, folder} which is now upload_{file, dir} and download_{file, dir}), making a ```_parse_url(s3_url)``` private function that parses the given url into ```bucket_name``` and ```prefix``` instead of us having to do it every time throughout our code, and formatting the input args (e.g., all functions that take in an s3 url or local path now are set to ```s3_url=...``` and ```local_path=...```.

## Final Update
Updated version formatting from 0.0 to 0.0.0 and are now starting over from 0.0.1, since we are replacing the PYPI package ```ml-dronebase-utils``` with ```ml-dronebase-data-utils```.

### CODE REVIEW

**Review by jonlwowski012 — APPROVED** [review]

File: ml_dronebase_data_utils/s3.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    split_prefix = os.path.join(parent_prefix, split, child_prefix)
+    return split_prefix
+
+
+def _split_labeled_dataset(
+    data_url: str,
+    labels_url: str,
+    train_split: int = 0.8,
+    val_split: Optional[float] = None,
+):
```
Comment (lnguyen11292021) [inline ml_dronebase_data_utils/s3.py:213]: Can you add some doc strings here? It looks like the function expects all label files in labels_url to have the same structures as data_url, but you have to read the code

File: ml_dronebase_data_utils/s3.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        split_prefix = _make_split_prefix(labels_prefix, "test")
+        move_files(labels_bucket_name, split_prefix, y_test)
+    else:
+        split_prefix = _make_split_prefix(data_prefix, "val")
+        move_files(data_bucket_name, split_prefix, x_val)
+        split_prefix = _make_split_prefix(labels_prefix, "val")
+        move_files(labels_bucket_name, split_prefix, y_val)
+
+
+def _split_unlabeled_dataset(
```
Comment (lnguyen11292021) [inline ml_dronebase_data_utils/s3.py:259]: This function looks very similar to the above except for the labels url, maybe there's a way to combine them?

File: ml_dronebase_data_utils/s3.py (hunk 1 above)
Comment (conorw8) [reply ml_dronebase_data_utils/s3.py:213]: This is an internal function which is called by the public function ```split_dataset```. The arguments are documented with doc strings in the public function.

File: ml_dronebase_data_utils/s3.py (hunk 2 above)
Comment (conorw8) [reply ml_dronebase_data_utils/s3.py:259]: I'd be open to suggestions, but if you combined them you would have an if/else block to check if you need to run ```x_train, x_val, y_train, t_val``` or just ```x_train, x_val``` and effectively repeat the code anyway. At least this way keeping the two functions separate keeps the two main ideas of splitting a labeled dataset and an unlabeled dataset separate.

**Review by lnguyen11292021 — APPROVED** [review]
